packages/dataviews/src/dataviews-layouts/utils/item-click-wrapper.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode, ReactElement, ComponentProps } from 'react';
+
+function getClickableItemProps< Item >( {
+	item,
+	isItemClickable,
+	onClickItem,
+	className,
+}: {
+	item: Item;
+	isItemClickable: ( item: Item ) => boolean;
+	onClickItem?: ( item: Item ) => void;
+	className?: string;
+} ) {
+	if ( ! isItemClickable( item ) || ! onClickItem ) {
+		return { className };
+	}
+
+	return {
+		className: className
+			? `${ className } ${ className }--clickable`
+			: undefined,
+		role: 'button',
```

Comment:
This is only needed in case no link component callback is provided?

Reply:
Ah sorry, I didn't see `getClickableItemProps` is only called for the default "link"